src/EditorFeatures/Core/CodeActions/CodeActionEditHandlerService.cs

```diff
@@ -284,7 +290,7 @@ internal class CodeActionEditHandlerService : ICodeActionEditHandlerService
                 }
 
                 _threadingContext.ThrowIfNotOnUIThread();
-                applied &= await operation.TryApplyAsync(workspace, progressTracker, cancellationToken).ConfigureAwait(true);
+                applied &= await operation.TryApplyAsync(workspace, originalSolution, progressTracker, cancellationToken).ConfigureAwait(true);
```

Comment:
this is one of the core places where we now ask the operation what change it would make, and we give it this additional information to make a better determination.